cluster/manager.go

```diff
@@ -46,26 +47,62 @@ var (
 	unmarshalErrMergeRemoteState = stats.NewCounter32("cluster.decode_err.merge_remote_state")
 )
 
-type ClusterManager struct {
+type ClusterManager interface {
+	IsPrimary() bool
+	SetPrimary(bool)
+	IsReady() bool
+	SetReady()
+	SetReadyIn(time.Duration)
+	SetState(NodeState)
+	ThisNode() Node
+	MemberList() []Node
+	Join([]string) (int, error)
+	GetPartitions() []int32
+	SetPartitions([]int32)
+	SetPriority(int)
+	Stop()
+	Start()
+}
+
+type MemberlistClusterManager struct {
```

Comment:
similar, no stuttering. so MemberlistManager

Reply:
this i can change